### PR TITLE
Various fixes for mlibc LFS part 3

### PR DESCRIPTION
**Pull request by Dennisbonke**
Round 3, mostly defines again, a few structs, two missing headers and several compatibility defines.

Part of the mlibc LFS project.

### CODE REVIEW

File: options/glibc/include/sys/kd.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef _SYS_KD_H
+#define _SYS_KD_H
+
+/* Make sure the <linux/types.h> header is not loaded.  */
```
Comment (64) [inline options/glibc/include/sys/kd.h:4]: Is this file copied from musl? The spacing should also be fixed to match mlibc's style.

File: options/posix/include/sys/syslog.h [hunk 2]
```diff
@@ -0,0 +1 @@
+#include <syslog.h>
```
Comment (64) [inline options/posix/include/sys/syslog.h:1]: This file should have a newline at the end

File: options/posix/include/dirent.h [hunk 3]
```diff
@@ -28,6 +28,8 @@ struct dirent {
 	char d_name[1024];
 };
 
+#define d_fileno d_ino
```
Comment (64) [inline options/posix/include/dirent.h:31]: Why is this needed? Seems like a bit of a hack

File: options/elf/include/elf.h [hunk 4]
```diff
@@ -144,6 +176,17 @@ __MLIBC_INLINE_DEFINITION unsigned char ELF64_ST_INFO(unsigned char bind, unsign
 	return (bind << 4) | type;
 }
 
+typedef struct {
+  Elf64_Half si_boundto;
+  Elf64_Half si_flags;
+} Elf64_Syminfo;
+
+#define ELF32_ST_BIND(val)		(((unsigned char) (val)) >> 4)
```
Comment (64) [inline options/elf/include/elf.h:184]: These should probably be made inline to match `ELF64_ST_*`.

**Review by 64 — CHANGES_REQUESTED** [review]

File: options/posix/include/dirent.h (hunk 3 above)
Comment (Dennisbonke) [reply options/posix/include/dirent.h:31]: It's a compatibility hack, used by `strace`. See [here](https://github.com/bminor/musl/blame/cfdfd5ea3ce14c6abf7fb22a531f3d99518b5a1b/include/dirent.h#L22) for what musl thought when adding it.

File: options/glibc/include/sys/kd.h (hunk 1 above)
Comment (Dennisbonke) [reply options/glibc/include/sys/kd.h:4]: Based on indeed, I'll fix the spacing.

File: options/elf/include/elf.h (hunk 4 above)
Comment (Dennisbonke) [reply options/elf/include/elf.h]: No longer in this PR, moved to #547

File: options/posix/include/sys/syslog.h [hunk 5]
```diff
@@ -0,0 +1,2 @@
...
+
```
Comment (64) [inline options/posix/include/sys/syslog.h:2]: There is one newline too many :^) otherwise LGTM

File: options/linux/include/sys/mount.h [hunk 6]
```diff
@@ -13,6 +13,8 @@ extern "C" {
 #define MS_SYNCHRONOUS 32
 #define MS_NOATIME 64
 
+#define MNT_FORCE 1
```
Comment (ArsenArsen) [inline options/linux/include/sys/mount.h:16]: is this ABI?

Comment (Dennisbonke) [reply options/linux/include/sys/mount.h:16]: As far as I know it isn't (and if it is, it matches Linux now).

Comment (ArsenArsen) [reply options/linux/include/sys/mount.h:16]: this isn't consumed by the libc, so it is OS ABI

Comment (64) [reply options/linux/include/sys/mount.h:16]: > As far as I know it isn't (and if it is, it matches Linux now).

Arsen meant that the change should be in `abis`, not that the change is an ABI break

**Review by 64 — APPROVED** [review]

File: options/glibc/include/sys/kd.h (hunk 1 above)
Comment (Geertiebear) [reply options/glibc/include/sys/kd.h:4]: What is this actually used for?

Comment (Dennisbonke) [reply options/glibc/include/sys/kd.h:4]: As far as I'm aware, this is just some glibc compatibility header.

Comment (Geertiebear) [reply options/glibc/include/sys/kd.h:4]: I meant the "Make sure linux/types.h is not loaded". What is that used for? Our ``linux/kd.h`` does not have any inclusion of ``linux/types.h``.

File: abis/mlibc/mount.h [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#ifndef _ABIBITS_MOUNT_H
+#define _ABIBITS_MOUNT_H
+
+#define MNT_FORCE 1
+
+#endif // _ABIBITS_MOUNT_H
```
Comment (avdgrinten) [inline abis/mlibc/mount.h:6]: Why does mlibc get its own ABI even though it matches the Linux ABI anyway?

**Review by avdgrinten — CHANGES_REQUESTED** [review]

File: abis/mlibc/mount.h (hunk 7 above)
Comment (Geertiebear) [reply abis/mlibc/mount.h:6]: For consistency's sake I suppose.

File: options/glibc/include/sys/kd.h [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+#define _LINUX_TYPES_H 1
+#define __undef_LINUX_TYPES_H
+#endif
+
+#include <linux/kd.h>
+
+#ifdef __undef_LINUX_TYPES_H
+#undef _LINUX_TYPES_H
+#undef __undef_LINUX_TYPES_H
+#endif
```
[the remark this answers is not in the document]
Comment (ArsenArsen) [reply options/glibc/include/sys/kd.h:15]: It's also not necessary, and should be in the Linux option I think. See: https://github.com/ArsenArsen/mlibc/commit/27d1035f368373a07641cbc532e3208cf6b77dd1

File: abis/mlibc/mount.h (hunk 7 above)
Comment (avdgrinten) [reply abis/mlibc/mount.h:6]: That doesn't make sense, sysdeps can just link to `abis/linux` instead of `abis/mlibc`.

**Comment by Dennisbonke** [comment]
Dropped the two commits giving issues, I'll split those out.

**Review by 64 — APPROVED** [review]

**Review by Geertiebear — APPROVED** [review]
Since the PR slimme down a bunch, the left over changes are good to merge
